base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```
@@ -332,15 +332,40 @@ impl LMDBDatabase {
                     block_accum_data.deleted = DeletedBitmap { deleted };
                     self.update_block_accumulated_data(&write_txn, height, &block_accum_data)?;
                 },
+                UpdateDeletedBlockAccumulatedDataWithDiff {
+                    header_hash,
+                    mut deleted,
+                } => {
+                    let height = self.fetch_height_from_hash(&write_txn, &header_hash).or_not_found(
+                        "BlockHash",
+                        "hash",
+                        header_hash.to_hex(),
+                    )?;
+                    let prev_block_accum_data = if height > 0 {
+                        self.fetch_block_accumulated_data(&write_txn, height - 1)?
+                            .unwrap_or_else(BlockAccumulatedData::default)
+                    } else {
+                        BlockAccumulatedData::default()
```

nit: Probably better to throw an Error (Attempt to update genesis block)